FinnUI/Sources/Components/ImageLinkView/ImageLinkView.swift

```diff
@@ -0,0 +1,158 @@
+        let imageView = RemoteImageView(withAutoLayout: true)
+        imageView.contentMode = .scaleAspectFill
+        imageView.clipsToBounds = true
+        imageView.layer.cornerRadius = .spacingS
+        imageView.isUserInteractionEnabled = true
+        imageView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleTap)))
+        return imageView
+    }()
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        addSubview(descriptionLabel)
+        addSubview(imageView)
+        addSubview(overlayView)
+
+        NSLayoutConstraint.activate(
+        [
+            descriptionLabel.topAnchor.constraint(equalTo: topAnchor),
+            descriptionLabel.leadingAnchor.constraint(equalTo: leadingAnchor),
+            descriptionLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+
+            imageView.topAnchor.constraint(equalTo: descriptionLabel.bottomAnchor, constant: .spacingS),
+            imageView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            imageView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            imageView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: 3/4),
+
+            overlayView.centerXAnchor.constraint(equalTo: imageView.centerXAnchor),
+            overlayView.centerYAnchor.constraint(equalTo: imageView.centerYAnchor)
+        ])
+    }
+
+    // MARK: - Public methods
+
+    public func configure(with viewModel: ImageLinkViewModel) {
+        self.viewModel = viewModel
+        descriptionLabel.text = viewModel.description
+        imageView.loadImage(for: viewModel.imageUrl, imageWidth: imageView.frame.width, loadingColor: viewModel.loadingColor, fallbackImage: viewModel.fallbackImage)
+
+        switch viewModel.overlayKind {
+        case .video?:
+            overlayView.configure(withImage: UIImage(named: .overlayVideo))
+            overlayView.isHidden = false
+        case .virtualViewing?:
+            overlayView.configure(withImage: UIImage(named: .overlayVirtualViewing))
+            overlayView.isHidden = false
+        default:
+            overlayView.isHidden = true
+        }
+    }
+
+    // MARK: - Actions
+
+    @objc private func handleTap() {
+        guard let url = viewModel?.url else { return }
+        delegate?.imageLinkViewWasSelected(self, url: url)
+    }
+}
+
+// MARK: - Private class
+
+private class OverlayView: UIView {
+
+    // MARK: - Private properties
+
+    private lazy var iconImageView: UIImageView = {
+        let imageView = UIImageView(withAutoLayout: true)
+        imageView.contentMode = .scaleAspectFit
+        imageView.tintColor = .milk
```

Comment:
is this correct for both light and dark modes?

Reply:
Yes :) The images won't change between light- or dark-mode, so this is correct for both.